### PR TITLE
feat(FEC-11761): expose stream timed metadata - phase 2

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes

* new type `TimedMetadata` instead of `CuePoint`
* new `TIMED_METADATA_CHANGE` event which triggered when with `TIMED_METADATA` but with payload of `TimedMetadata` array instead of `VttCue` array
* remove `track.label` and merge all metadata tracks cues instead (reverts https://github.com/kaltura/playkit-js/pull/609, closes https://github.com/kaltura/playkit-js/pull/610)
* new function `createTimedMetadata` to create a `TimedMetadata` based on `VttCue`
* new helper function `binarySearch` (for https://github.com/kaltura/playkit-js-hls/pull/157)

Solves FEC-11761
Related to - 
https://github.com/kaltura/playkit-js-hls/pull/157 
https://github.com/kaltura/playkit-js-dash/pull/174
https://github.com/kaltura/kaltura-player-js/pull/512
https://github.com/kaltura/playkit-js-dual-screen/pull/52
https://github.com/kaltura/playkit-js-kaltura-cuepoints/pull/15

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [x] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/track/cue-point.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ class CuePoint {
 CuePoint.TYPE = {
   ID3: 'id3',
   EMSG: 'emsg',
+  CUE_POINT: 'cuepoint',
```
Comment (RoyBregman) [inline src/track/cue-point.js:30]: CuePount.Type.CUE_POINT?

File: src/engines/html5/html5.js [hunk 2]
```diff
@@ -1166,24 +1168,44 @@ export default class Html5 extends FakeEventTarget implements IEngine {
     const listenToCueChange = track => {
       track.mode = PKTextTrack.MODE.HIDDEN;
       this._eventManager.listen(track, 'cuechange', () => {
-        this.dispatchEvent(new FakeEvent(CustomEventType.TIMED_METADATA, {cues: Array.from(track.activeCues), label: track.label}));
+        let activeCues = [];
+        Array.from(this._el.textTracks).forEach((track: TextTrack) => {
```
Comment (RoyBregman) [inline src/engines/html5/html5.js:1172]: there are two "track" props nested. Maybe change the one in line 1168 to "changedTrack" or something to differ

File: src/utils/binary-search.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  const mid = Math.floor(list.length / 2);
+  if (comparisonFn(list[mid]) === 0) {
+    return list[mid];
+  }
+  if (comparisonFn(list[mid]) > 0) {
+    return binarySearch(list.slice(0, mid), comparisonFn);
+  } else {
+    return binarySearch(list.slice(mid + 1), comparisonFn);
+  }
+}
```
Comment (RoyBregman) [inline src/utils/binary-search.js:26]: What does this have to do with this feat?

Comment (yairans) [reply src/utils/binary-search.js:26]: see https://github.com/kaltura/playkit-js-hls/pull/157/files#diff-79225314d830d5c675af1144ac9cc91c4f1589ec811160895dc6b5138be20646R452

File: src/track/cue-point.js (hunk 1 above)
Comment (yairans) [reply src/track/cue-point.js:30]: agree it's wired but this is how cuepoint manager works. it generates a cue point of type 'cuepoint'.

File: src/engines/html5/html5.js (hunk 2 above)
Comment (yairans) [reply src/engines/html5/html5.js:1172]: done
